elyra/airflow/bootstrapper.py

```
@@ -201,6 +209,44 @@ def has_wildcard(self, filename):
+        if self.has_wildcard(path):
+            # Extract the directory component before the wildcard
+            dir_part = os.path.dirname(path)
+            if dir_part:
+                # Resolve and validate the directory part
+                try:
+                    resolved_dir = os.path.realpath(os.path.join(base_dir, dir_part))
+                    if not resolved_dir.startswith(
+                        os.path.realpath(base_dir) + os.sep
+                    ) and resolved_dir != os.path.realpath(base_dir):
+                        raise ValueError(f"Path traversal detected: {path} resolves outside working directory")
+                except Exception:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)